cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/Attachment.java

```diff
@@ -128,9 +112,6 @@ public boolean equals(Object o) {
 
         Attachment that = (Attachment) o;
 
-        if (name != null ? !name.equals(that.name) : that.name != null) {
-            return false;
-        }
```

Comment:
This method is now basically saying that `Attachment`s are equal if they have the same type and encoding. That's a pretty loose equality! A few possibilities spring to mind although I'm sure there are others...
1. Use default `equals` & `hashcode` in this class, and override in subclasses where a more useful check can be made.
1. Leave these here and override in subclasses, calling `super` to get additional part of the comparisons. The downside of this is that we need to remember.
1. Add `abstract` methods that subclasses are forced to implement that is used by this `equals` and `hashcode` to provides the additional parts whilst declaring this `equals` and `hashcode` `final`. This is rigorous about ensuring correct implementations, but maybe a bit weird.
1. Decide that we aren't going to compare `Attachment` equality or use them as keys (since we tend to key of the `String` name so we just use the default `Object.equals` & `hashcode`.